plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -894,9 +894,13 @@ public void testCreatePartitionedTableWithNestedTypes()
     @Test
     public void testCreatePartitionedTableAs()
     {
+        File tempDir = this.getDistributedQueryRunner().getCoordinator().getBaseDataDir().toFile();
```

Comment:
nit `this.` is redundant

also, some directory under `tempDir`, like `"test_create_partitioned_" + randomTableSuffix()`, to avoid writing to (and potentially messing) in base data dir directly.